konan_sdk/endpoints/konan_endpoints.py

```diff
@@ -1,52 +1,121 @@
-from konan_sdk.endpoints.base_endpoint import KonanBaseEndpoint
+from typing import Any, Dict, List, Union
+from konan_sdk.endpoints.base_endpoint import KonanBaseEndpoint, KonanBaseDeploymentEndpoint
+from konan_sdk.endpoints.interfaces import KonanEndpointRequest, KonanEndpointResponse
+from konan_sdk.konan_types.konan_feedback import FeedbackStatus, FeedbackSubmission
 
 
 class LoginEndpoint(KonanBaseEndpoint):
-    name = 'login'
-
     @property
-    def endpoint_path(self):
-        return '/api/auth/login/'
+    def name(self) -> str:
+        return 'login'
 
     @property
-    def headers(self):
-        return {}
+    def endpoint_path(self) -> str:
+        return '/api/auth/login'
 
+    class RequestObject():
+        def __init__(self, email: str, password: str) -> None:
+            self.email = email
+            self.passord = password
 
-class RefreshTokenEndpoint(KonanBaseEndpoint):
-    name = 'refresh_token'
+    class ResponseObject():
+        def __init__(self, access_token: str, refresh_token: str) -> None:
+            self.access_token = access_token
+            self.refresh_token = refresh_token
+
+    def prepare_request(self, request_object: RequestObject) -> KonanEndpointRequest:
+        return KonanEndpointRequest(data={'email': request_object.email, 'password': request_object.passord})
```

Comment:
Hmm, I can get behind the idea of `KonanEndpointResponse` and `ResponseObject` but I am not sure if I totally agree with the `RequestObject` idea as to me it feels like an unnecessary class to pass things. I propose instead we do it the old way but a bit different like:

In `KonanBaseEndpoint`:
```python
    @abstractmethod
    def prepare_request(self, payload: Dict[str, Any], params: Dict[str, Any]) -> KonanEndpointRequest:
        # Override method with logic to implement before sending request
        return KonanEndpointRequest(data=payload, params=params)
        
    def post(self, payload: Dict[str, Any], params: Dict[str, Any]) -> Any:
        endpoint_request = self.prepare_request(payload=payload, params=params)
 ```
 Let me know what you think

Reply:
After reading your comments, I refactored all endpoints to inherit from a `KonanBaseEndpoint` with `Generic` types for request objects and response objects. Each child `*Endpoint` would specify the exact types it expects to use when creating POST/GET request and when reading such a request's result.

The motivation behind having those two types be dynamic per endpoint (and not simply rely on a `Dict`) is to help both future `konan-sdk` developers and end users consuming those APIs.

Let me know if the updated implementation clears up your concerns or not. I'm happy to modify it further a bit if you still feel the flow is not smooth or simple enough.

Reply:
I like it, it might seem a bit too generic/complex but perhaps that is only at first glance and the benefits will be evident later